File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    let currentPointer = head

    while(currentPointer !== null ) {
        let nextPointer = currentPointer.next

        while(nextPointer !== null && nextPointer.val === currentPointer.val) {
            nextPointer = nextPointer.next
        }

        currentPointer.next = nextPointer
        currentPointer = nextPointer
    }
    return head
};

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
    currentPointer := head
    
    for currentPointer != nil  {
        nextPointer := currentPointer.Next 
        
        for nextPointer != nil && nextPointer.Val == currentPointer.Val {
            nextPointer = nextPointer.Next
        }
        currentPointer.Next = nextPointer
        currentPointer = nextPointer
    }
    return head
}